controllers/gitrepository_controller.go

```diff
@@ -751,11 +755,7 @@ func (r *GitRepositoryReconciler) gitCheckout(ctx context.Context,
 		}
 	}
 
-	// Checkout HEAD of reference in object
-	gitCtx, cancel := context.WithTimeout(ctx, obj.Spec.Timeout.Duration)
-	defer cancel()
-
-	commit, err := checkoutStrategy.Checkout(gitCtx, dir, obj.Spec.URL, authOpts)
```

Comment:
In this approach, all the callers of `gitCheckout()` will have to add such timeouts. We can do even better by just moving this gitCtx right above https://github.com/fluxcd/source-controller/blob/82cd05eb3d0b8f4c222b99a3cc9943e5790982f7/controllers/gitrepository_controller.go#L724.
That way, it's simple for the users of the functions and the cancellation is confined to this function.

Reply:
Okie dokie, changed accordingly. :+1: